### PR TITLE
Converting Inheritance relationship tests to AssertQuery infra

### PR DESCRIPTION
**Pull request by maumar**

### CODE REVIEW

File: test/EFCore.Specification.Tests/TestModels/InheritanceRelationshipsModel/DerivedCollectionOnBase.cs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-namespace Microsoft.EntityFrameworkCore.TestModels.InheritanceRelationships
+namespace Microsoft.EntityFrameworkCore.TestModels.InheritanceRelationshipsModel
```
Comment (smitpatel) [inline test/EFCore.Specification.Tests/TestModels/InheritanceRelationshipsModel/DerivedCollectionOnBase.cs:4]: Isn't Model in the name redundant given they are in TestModels namespace already?

File: test/EFCore.Specification.Tests/Query/InheritanceRelationshipsQueryTestBase.cs [hunk 2]
```diff
@@ -73,454 +76,587 @@ public virtual void Entity_can_make_separate_relationships_with_base_type_and_de
...
+        [MemberData(nameof(IsAsyncData))]
+        public virtual Task Include_collection_with_inheritance_with_filter_reverse(bool async)
+        {
+            return AssertQuery(
+                async,
+                ss => ss.Set<BaseCollectionOnBase>().Include(e => e.BaseParent).Where(e => e.Name != "Bar"),
+                elementAsserter: (e, a) => AssertInclude(
+                    e,
+                    a,
+                    new ExpectedInclude<BaseCollectionOnBase>(x => x.BaseParent)));
```
Comment (smitpatel) [inline test/EFCore.Specification.Tests/Query/InheritanceRelationshipsQueryTestBase.cs:258]:
```suggestion
                elementAsserter: (e, a) => AssertInclude(e, a, new ExpectedInclude<BaseCollectionOnBase>(x => x.BaseParent)));
```

File: test/EFCore.Relational.Specification.Tests/Query/InheritanceRelationshipsQueryRelationalTestBase.cs [hunk 3]
```diff
@@ -15,214 +17,272 @@ public InheritanceRelationshipsQueryRelationalTestBase(TFixture fixture)
...
-
-            Assert.Equal(2, result.Count);
+            return AssertQuery(
+                async,
+                ss => ss.Set<ReferencedEntity>().Include(e => e.Principals).ThenInclude(e => e.Reference),
+                elementAsserter: (e, a) => AssertInclude(
+                    e,
+                    a,
+                    new ExpectedInclude<ReferencedEntity>(x => x.Principals),
+                    new ExpectedInclude<PrincipalEntity>(x => x.Reference)));
```
Comment (smitpatel) [inline test/EFCore.Relational.Specification.Tests/Query/InheritanceRelationshipsQueryRelationalTestBase.cs:285]:
```suggestion
                elementAsserter: (e, a) => AssertInclude(e, a,
                    new ExpectedInclude<ReferencedEntity>(x => x.Principals),
                    new ExpectedInclude<PrincipalEntity>(x => x.Reference)));
```

File: test/EFCore.Specification.Tests/TestModels/InheritanceRelationshipsModel/DerivedCollectionOnBase.cs (hunk 1 above)
Comment (maumar) [reply test/EFCore.Specification.Tests/TestModels/InheritanceRelationshipsModel/DerivedCollectionOnBase.cs:4]: did it for consistency, we could switch them all to non-Model, but thats not the only redundancy in out test naming ;)

**Pull request merged**